### PR TITLE
Fix handling of os signals in tink server

### PR DESCRIPTION
**Pull request by krishnanarchana**
Related to #527

## Description

Previously tink server failed to respect os signals like SIGTERM,SIGINT etc.

This PR fixes the tink server so that os signals sent to the tink server process
will actually kill the process.

## Why is this needed

Sending os signals actually terminates the tink server.

Fixes: #527

## How Has This Been Tested?
- Tested using docker-compose:

```bash
make images
make run
docker exec -it tink_tinkerbell_1 sh
ps
kill -SIGINT 1
```

Corresponding logs:

```
tinkerbell_1             | {"level":"info","ts":1630442128.6128566,"caller":"tink-server/main.go:198","msg":"signal received, stopping servers","service":"github.com/tinkerbell/tink","signal":"interrupt"}
tinkerbell_1             | {"level":"info","ts":1630442129.4250214,"caller":"metrics/metrics.go:58","msg":"initializing label values","service":"github.com/tinkerbell/tink"}
tinkerbell_1             | {"level":"info","ts":1630442129.4286315,"caller":"http-server/http_server.go:88","msg":"serving http","service":"github.com/tinkerbell/tink"}
```

Now the docker container exists and we can see the log that signal is received.

Signed-off-by: Archana krishnan <krisharchana9@gmail.com>

### CODE REVIEW

**Comment by jacobweinstock** [comment]
Hey @krishnanarchana! Welcome and thank you for this contribution!

Looks like you'll need the DCO sign-off before we can merge this. Here's a [guide](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) to help with that if needed.

**Review by jacobweinstock — DISMISSED** [review]

**Comment by krishnanarchana** [comment]
Hi @jacobweinstock  thank you for your review, will do the DCO signoff.

**Comment by krishnanarchana** [comment]
> Hi @jacobweinstock thank you for your review, will do the DCO signoff.

@jacobweinstock completed the DCO signoff. Thanks!

**Review by jacobweinstock — APPROVED** [review]

**Pull request merged**
